### ISSUE TITLE
Add System.Devices.* APIs to Microsoft.Windows.Compatibility pack

### ISSUE THREAD

**Issue by dotMorten** [dotnet/corefx#39918]
The System.Devices APIs are not part of the compatibility pack, nor are these APIs open-sourced for anyone else to duplicate the functionality in this library.

Please consider adding these to the Microsoft.Windows.Compatibility NuGet package for 3.0 to reduce the API gap when moving from .NET Framework to .NET Core.

**Comment by joperezr** [comment]
cc: @safern and @ericstj who have worked with the compatibility package before.

@dotMorten I believe the compatibility package is really only facades that would help reroute types from one framework to another, so for that to work here System.Devices apis that you are interested in should already be open sourced. Also, I suppose you mean `System.Device.Location` and not System.Devices?

**Comment by dotMorten** [comment]
@joperezr The DLL reference is System.Devices, and yes it mainly contains the location APIs.

> System.Devices apis that you are interested in should already be open sourced

Where? It's not even on referencesource.

**Comment by joperezr** [comment]
> The DLL reference is System.Devices, and yes it mainly contains the location APIs.

From looking here https://apisof.net/catalog/System.Device.Location seems like they are in System.Device.dll version 4.0 starting from .NET 4.0

> Where? It's not even on referencesource.

Right, sorry I didn't write what I intended. What I meant is that in order for us to include them in the compatibility pack, we first need to go an open source them and ship them for .NET Core via a NuGet package, so that needs to happen first, and then we can add them to the compat pack. Is there a specific type that you are interested in? (Asking since we may already have a different type in both .NET and .NETCore that might satisfy your needs.)

**Comment by dotMorten** [comment]
The ones I'm specifically interested in are the GeocoordinateWatcher and all accompanying types.
I know Win10 has similar APIs (Geolocator) but that create a hard dependency on Win10.

**Comment by safern** [comment]
cc: @danmosemsft who might have more context why these APIs were left out of the porting effort.

**Comment by ericstj** [comment]
Looks to be built upon the Windows Location COM API https://docs.microsoft.com/en-us/windows/win32/locationapi/windows-location-api-portal.  Looks like this API is deprecated and not even supported on many Windows SKUs.  This issue and https://github.com/dotnet/corefx/issues/23393 appear to be duplicates.

**Comment by safern** [comment]
Closing in favor of: #23393
